src/main/java/com/revlearn/team1/mapper/TransactionMapper.java

```diff
@@ -6,20 +6,24 @@
 import com.revlearn.team1.model.Course;
 import com.revlearn.team1.model.TransactionModel;
 import com.revlearn.team1.model.User;
+import com.revlearn.team1.repository.UserRepository;
 import com.revlearn.team1.service.user.UserService;
+import com.revlearn.team1.service.user.UserServiceImp;
 import jakarta.persistence.EntityNotFoundException;
 import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 import java.util.List;
+import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
 @Component
 @RequiredArgsConstructor
 public class TransactionMapper 
 {
-
-    private final UserService userService;
+    @Autowired
+    private final UserServiceImp userService;
```

Comment:
should be using the UserService interface instead of the implementation for best practices! Also I don't think the autowired is necessary since this is the only userService so spring boot will auto use it

Reply:
It is the @RequiredArgsConstructor annotation that allows the exclusion of the @Autowired annotation.

Reply:
oh okie!